### PR TITLE
VCDA-4001 rde storage class

### PR DESCRIPTION
**Pull request by abaruni**
## Description
Please provide a brief description of the changes proposed in this Pull Request

- update vcdcluster status to include storage class options

## Checklist
- [x] tested locally
- [ ] updated any relevant dependencies
- [ ] updated any relevant documentation or examples

## API Changes
Are there API changes?
- [x] Yes
- [ ] No

If yes, please fill in the below

1. Updated conversions?
    - [x] Yes
    - [ ] No
    - [ ] N/A
2. Updated CRDs?
    - [x] Yes
    - [ ] No
    - [ ] N/A
3. Updated infrastructure-components.yaml?
    - [x] Yes
    - [ ] No
    - [ ] N/A
4. Updated `./examples/capi-quickstart.yaml`?
    - [ ] Yes
    - [ ] No
    - [x] N/A
5. Updated necessary files under `./infrastructure-vcd/v1.0.0/`?
   - [x] Yes
   - [ ] No
   - [ ] N/A

## Issue
If applicable, please reference the relevant issue

Fixes #

---
This change is [Reviewable](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/201)

### CODE REVIEW

**Review by arunmk — APPROVED** [review]
:lgtm: with the only question on using kubebuilder annotations

> Reviewed 1 of 7 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/201#-:-N6Twui5-aTRAiRVeRAs:b-xdck4r)* status: 1 of 7 files reviewed, 2 unresolved discussions (waiting on @abaruni and @arunmk)

___
*[`api/v1beta1/vcdcluster_types.go` line 141 at r1](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/201#-N6TtvRiBjovXnCxKY-v:-N6TtvRiBjovXnCxKY-w:byytuv3) ([raw file](https://github.com/vmware/cluster-api-provider-cloud-director/blob/fc20f3bab56e41a737404a75caa550c5101ecfe8/api/v1beta1/vcdcluster_types.go#L141)):*
> ```Go
> 
> 	// +optional
> 	DefaultStorageClassOptions DefaultStorageClassOptions `json:"defaultStorageClassOptions"`
> ```

An `omitempty` is useful when the field is optional. Kubebuilder does not truly treat fields as optional unless that is set
___
*[`config/crd/bases/infrastructure.cluster.x-k8s.io_vcdclusters.yaml` line 301 at r1](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/201#-N6TuSBA8eMGJ94Nzkqb:-N6TuSBA8eMGJ94Nzkqc:b-97u7zz) ([raw file](https://github.com/vmware/cluster-api-provider-cloud-director/blob/fc20f3bab56e41a737404a75caa550c5101ecfe8/config/crd/bases/infrastructure.cluster.x-k8s.io_vcdclusters.yaml#L301)):*
> ```YAML
>                   type: object
>                 type: array
>               defaultStorageClassOptions:
> ```

I don't remember the defaults but @ltimothy7 knows and he should review this part. Especially the `useDeleteReclaimPolicy` which was discussed a lot.

On a separate note this is generated code, correct? We should use the `+kubebuilder` annotations to specify the defaults.

**Review by sahithi — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/201)* status: 0 of 7 files reviewed, 1 unresolved discussion (waiting on @abaruni and @sahithi)

___
*[`controllers/vcdcluster_controller.go` line 249 at r2](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/201#-N6USAr29YBiiSTx7p9R:-N6USAr29YBiiSTx7p9S:b-n9rp6j) ([raw file](https://github.com/vmware/cluster-api-provider-cloud-director/blob/4c24422256e170e044f6ccc03a5f7d734d8311d3/controllers/vcdcluster_controller.go#L249)):*
> ```Go
> 				CapiStatusYaml:             "",
> 				ClusterResourceSetBindings: nil,
> 				DefaultStorageClass: rdeType.DefaultStorageClass{
> ```

defaultStorageClass options can only be applied at the time of cluster creation.
If the user changes the spec values later, incorrect values will begin to show up in the RDE and vcdcluster.status.

We need to do something like 



_Code snippet:_
```Go
if vcdclusterStatus is not empty , then update it from spec, else skip.

RDE.status.capvcd.defaultStorageClassOptions need to be updated only from vcdCluster.status
```

**Review by sahithi — APPROVED** [review]
:lgtm: after addressing the comment 

> *[Reviewable](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/201#-:-N6UU9jS4GLKcz3XUf4U:b-3tyik1)* status: 0 of 7 files reviewed, 1 unresolved discussion (waiting on @abaruni and @sahithi)

**Pull request merged**
